File: 883/projection-area-of-3d-shapes.go
package main

import "fmt"

func projectionArea(grid [][]int) int {
	var top, front, side int
	for i := range grid {
		x, y := 0, 0
		for j := range grid {
			if grid[i][j] != 0 {
				top += 1
			}
			x = max(x, grid[i][j])
			y = max(y, grid[j][i])
		}
		front += x
		side += y
	}
	return top + front + side
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	grid := [][]int{
		{2},
	}
	fmt.Println(projectionArea(grid))
}

File: 883/projection-area-of-3d-shapes.py
class Solution():
    def projectionArea(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        top, front, side = 0, 0, 0
        n = len(grid)
        for i in range(n):
            x, y = 0, 0
            for j in range(n):
                if grid[i][j] != 0:
                    top += 1
                x = max(x, grid[i][j])
                y = max(y, grid[j][i])
            
            front += x
            side += y
        
        return top + front + side
        